bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -111,15 +139,33 @@ private SecurityContext authenticateBearerToken(String token) throws Authenticat
         }
     }
 
-    private SecurityContext authenticateUsernamePassword(String username, String password)
+    private SecurityContext authenticateUsernamePassword(String username, String password, String authCacheKey)
             throws AuthenticationException {
+        final ByteBuffer cacheKey = getCacheKey(authCacheKey);
+        final UserSecurityContext cachedValue = AuthFilter.authCache.get(cacheKey);
+        if (cachedValue != null) {
+            return cachedValue;
+        }
+
         UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(username, password);
         Authentication auth = userRegistry.authenticate(credentials);
         User user = userRegistry.get(auth.getUsername());
         if (user == null) {
             throw new AuthenticationException("User not found in registry");
         }
-        return new UserSecurityContext(user, auth, "Basic");
+
+        // limit cache to a certain size
+        if (AuthFilter.authCache.size() > 20) {
```

Comment:
Magic numbers like "20" should be defined as a constant.
20 feels pretty high for me as well.

Reply:
Okay - sure thing. 
About the size: keep in mind the cache can also contain invalid auth attempts.
What would be a better number? 10?

Reply:
Ok, good point, I didn't think of the invalid attempts. But yeah, 10 should be enough, I'd say.